platform/utils/canonical_posix.go

```diff
@@ -0,0 +1,21 @@
+//go:build unix
+
+package utils
+
+// #include <stdlib.h>
+import "C"
+import (
+	"unsafe"
+)
+
+func Canonical(path string) (result string, err error) {
```

Comment:
Could you please explain why C calls are needed? Are there any alternatives?

Reply:
Also how did you come up with such solution, is it recommeded?

Reply:
Because that would not convert case into canonical form.

Reply:
Please add it as a doc to this function and rename it to `CanonicalPath` so we can understand it's about paths

Reply:
I meant in the doc add motivation why go sdk is not enough and we have to use C

Reply:
Added more comments